src/Input/Select/Select.numberChecking.test.tsx

```diff
@@ -0,0 +1,123 @@
+  test('number option children, click option', async () => {
+    const onChange = jest.fn();
+
+    const Component = () => (
+      <Select label="number option children" onChange={onChange}>
+        <Select.Option value="one">{1}</Select.Option>
+        <Select.Option value="two">{2}</Select.Option>
+        <Select.Option value="three">{3}</Select.Option>
+      </Select>
+    );
+
+    const { queryByText, asFragment } = render(<Component />);
+
+    const downArrowIcon = document.querySelector('.select-icon');
+    fireEvent.click(downArrowIcon);
+    const option1 = queryByText('1');
+    fireEvent.mouseDown(option1);
+
+    expect(onChange).toHaveBeenLastCalledWith('one');
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('number option children, search filtering', async () => {
+    const Component = () => (
+      <Select label="number option children">
+        <Select.Option value="one">{1}</Select.Option>
+        <Select.Option value="two">{2}</Select.Option>
+        <Select.Option value="three">{3}</Select.Option>
+      </Select>
+    );
+
+    const { queryByRole, queryAllByTestId, asFragment } = render(<Component />);
+
+    const selectInput = queryByRole('combobox');
+    userEvent.type(selectInput, '1');
+    const filteredOptions = queryAllByTestId('option');
+    expect(filteredOptions.length).toEqual(1);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('number defaultValue', async () => {
+    const testNumberDefaultValue = ([defaultValue, expectedValue]: any) => {
+      const Component = () => (
+        <Select label="number defaultValue" defaultValue={defaultValue}>
+          <Select.Option value="one">{1}</Select.Option>
+          <Select.Option value="two">{2}</Select.Option>
+          <Select.Option value="three">{3}</Select.Option>
+        </Select>
+      );
+
+      const { queryByRole, unmount } = render(<Component />);
+      const selectInput = queryByRole('combobox') as HTMLInputElement;
+      expect(selectInput.value).toEqual(expectedValue);
+      unmount();
+    };
+
+    [
+      [1, '1'],
+      [{ key: 'value' }, '[object Object]'],
+      [[1, 2, 3], '1,2,3'],
+    ].forEach(([defaultValue, expectedValue]) => {
+      testNumberDefaultValue([defaultValue, expectedValue]);
+    });
+  });
+
+  test('number value', async () => {
+    const testNumberValue = ([value, expectedValue]: any) => {
+      const Component = () => (
+        <Select label="number defaultValue" value={value}>
+          <Select.Option value="one">{1}</Select.Option>
+          <Select.Option value="two">{2}</Select.Option>
+          <Select.Option value="three">{3}</Select.Option>
+        </Select>
+      );
+
+      const { queryByRole, unmount } = render(<Component />);
+      const selectInput = queryByRole('combobox') as HTMLInputElement;
+      expect(selectInput.value).toEqual(expectedValue);
+      unmount();
+    };
+
+    [
+      [1, '1'],
+      [{ key: 'value' }, '[object Object]'],
```

Comment:
same comment for this one.
https://github.com/glints-dev/glints-aries/pull/996/files#r1424903960